plugins/source/azure/resources/services/monitor/diagnostic_settings.go

```diff
@@ -1,281 +1,130 @@
-						Name:        "retention_policy_enabled",
-						Description: "a value indicating whether the retention policy is enabled",
-						Type:        schema.TypeBool,
-						Resolver:    schema.PathResolver("RetentionPolicy.Enabled"),
-					},
-					{
-						Name:        "retention_policy_days",
-						Description: "the number of days for the retention in days A value of 0 will retain the events indefinitely",
-						Type:        schema.TypeInt,
-						Resolver:    schema.PathResolver("RetentionPolicy.Days"),
-					},
-				},
+				Name:     "name",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Name"),
 			},
 			{
-				Name:        "azure_monitor_diagnostic_setting_logs",
-				Description: "LogSettings part of MultiTenantDiagnosticSettings Specifies the settings for a particular log",
-				Resolver:    fetchMonitorDiagnosticSettingLogs,
-				Columns: []schema.Column{
-					{
-						Name:        "diagnostic_setting_cq_id",
-						Description: "Unique ID of azure_monitor_diagnostic_settings table (FK)",
-						Type:        schema.TypeUUID,
-						Resolver:    schema.ParentIdResolver,
-					},
-					{
-						Name:        "diagnostic_setting_id",
-						Description: "Unique ID of azure_monitor_diagnostic_settings table (FK)",
-						Type:        schema.TypeString,
-						Resolver:    schema.ParentResourceFieldResolver("id"),
-					},
-					{
-						Name:        "category",
-						Description: "Name of a Diagnostic Log category for a resource type this setting is applied to To obtain the list of Diagnostic Log categories for a resource, first perform a GET diagnostic settings operation",
-						Type:        schema.TypeString,
-					},
-					{
-						Name:        "enabled",
-						Description: "a value indicating whether this log is enabled",
-						Type:        schema.TypeBool,
-					},
-					{
-						Name:        "retention_policy_enabled",
-						Description: "a value indicating whether the retention policy is enabled",
-						Type:        schema.TypeBool,
-						Resolver:    schema.PathResolver("RetentionPolicy.Enabled"),
-					},
-					{
-						Name:        "retention_policy_days",
-						Description: "the number of days for the retention in days A value of 0 will retain the events indefinitely",
-						Type:        schema.TypeInt,
-						Resolver:    schema.PathResolver("RetentionPolicy.Days"),
-					},
-				},
+				Name:     "type",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Type"),
 			},
 		},
 	}
 }
 
-// ====================================================================================================================
-//
-//	Table Resolver Functions
-//
-// ====================================================================================================================
-func fetchMonitorDiagnosticSettings(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	cl := meta.(*client.Client)
-	resSvc := cl.Services().Resources.Resources
```

Comment:
Instead of getting the resources here and running with multiple go routines, I created a new top level resource called `Resources` and made `DiagnosticSettings` it's relation. That way we can control the number of go routines in the SDK